stan/math/prim/core/complex_base.hpp

```diff
@@ -2,270 +2,232 @@
 #define STAN_MATH_PRIM_CORE_COMPLEX_BASE_HPP
 
 #include <stan/math/prim/fun/square.hpp>
+#include <stan/math/prim/meta.hpp>
 #include <complex>
 
 namespace stan {
 namespace math {
 
 /**
- * Base class for complex numbers.  The template parameter `V` is the
- * value type.  This class is used like the curiously recursive
- * template pattern (CRTP) with derived type `complex<V>`.  Thus any
- * extending class must be of the form `complex<V>`.
+ * Base class for complex numbers.  Extending classes must be of
+ * of the form `complex<ValueType>`.
  *
- * @tparam V value type for extending complex class
+ * @tparam ValueType type of real and imaginary parts
  */
-template <typename V>
+template <typename ValueType>
 class complex_base {
- protected:
-  /**
-   * Real part.
-   */
-  V re_;
-
-  /**
-   * Imaginary part.
-   */
-  V im_;
-
  public:
   /**
-   * Type of real and imaginary parts.
+   * Type of real and imaginary parts
    */
-  using value_type = V;
+  using value_type = ValueType;
```

Comment:
Is our standard to do underscore_naming for usings and CamelCase for template parameters?

Reply:
That's consistent with how I understand the code but it's weird to see if directly in a situation like this :/.